### DISCUSSION TITLE
How can I upgrade from master (3.7) to feature (4.0) branch?

### DISCUSSION THREAD

**Discussion by opericgithub** [netbox-community/netbox#15512] [Q&A]
When I was on a production (master) version 3.7.3 I was able to do the following:

```
cd /opt/netbox
sudo git pull origin feature
sudo ./upgrade.sh
```

and after that I was on new "alpha" 4.0.0-dev version.

Now, when I am on 3.7.4 I am not able to upgrade to 4.0.0-dev anymore.
I get these messages:

```
Running migrations:
  Applying dcim.0186_location_facility... OK
  Applying extras.0108_convert_reports_to_scripts... OK
  Applying extras.0109_script_model... OK
  Applying extras.0110_remove_eventrule_action_parameters... OK
  Applying extras.0111_rename_content_types...Traceback (most recent call last):
  File "/opt/netbox/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 103, in _execute
    return self.cursor.execute(sql)
  File "/opt/netbox/venv/lib/python3.10/site-packages/psycopg/cursor.py", line 732, in execute
    raise ex.with_traceback(None)
psycopg.errors.UndefinedTable: relation "extras_customfield_content_types_id_seq" does not exist
...
  File "/opt/netbox/venv/lib/python3.10/site-packages/psycopg/cursor.py", line 732, in execute
    raise ex.with_traceback(None)
django.db.utils.ProgrammingError: relation "extras_customfield_content_types_id_seq" does not exist

```
Is there some problem with the latest master version upgrading to feature?

**Comment by milevskiy2** [comment]
I have the same problem. 

I'm upgrading from 3.7.4 to 4.0.6 with commands: 

cd /opt/netbox
sudo git pull origin feature
sudo ./upgrade.sh

and get the error 

502 Bad Gateway
nginx/1.18.0 (Ubuntu)

**Reply by candlerb** [reply]
Don't do this. v4.0 is on the master branch now.

The feature branch is for developers only. If you ever update your instance to the "feature" branch, you will likely have to throw it away and lose any data you have added. Upgrades from feature to master, or even from feature to later feature, are not supported. Incompatible database schema changes can be made which mean it's impossible to move forward.

**Reply by milevskiy2** [reply]
Got it. Thanks.
